Adding Joi config checks to frontend

StanHus opened the pull request:
Relation: https://eaflood.atlassian.net/browse/FFAFGB-103

Adding checks for frontend using Joi

MohammedFoysal commented on app/config/index.js:
```diff
@@ -1,15 +1,82 @@
+const schema = Joi.object({
+  env: Joi.string().valid('development', 'test', 'production').required(),
+
+  fundingFarmingApiUri: Joi.string().uri().required(),
```
This env var is no longer used and can be removed. Remove it from here and the config on line 75

MohammedFoysal commented on app/config/index.js:
```diff
@@ -1,15 +1,82 @@
+  env: Joi.string().valid('development', 'test', 'production').required(),
+
+  fundingFarmingApiUri: Joi.string().uri().required(),
+  fundingFarmingApiKey: Joi.string().required(),
```
This one can be removed too

StanHus replied:
It seems to be used in 2 places actually. Still remove it?

![Screenshot 2024-06-24 at 14 34 48](https://github.com/DEFRA/fcp-find-ai-frontend/assets/86333618/412f626b-2477-4b12-b3a2-1ad94beade69)

MohammedFoysal replied:
Yes, if you can remove the usages too. We don't use this environment variable after switching to langchain. FARMING_FUNDING_API_URI can be removed too

StanHus replied:
done

MohammedFoysal commented on app/config/index.js:
```diff
@@ -1,15 +1,82 @@
+
+  fundingFarmingApiUri: Joi.string().uri().required(),
+  fundingFarmingApiKey: Joi.string().required(),
+  appInsightsKey: Joi.string().optional(),
```
This looks good. please edit the docker-compose.yaml file and remove the line where APPINSIGHTS_CONNECTIONSTRING is set then this check shouldn't give false positives

StanHus replied:
done

MohammedFoysal commented:
Looks good, just two changes please. Also the unit tests are failing resulting in a failing build - can you investigate?

StanHus commented:
> Looks good, just two changes please. Also the unit tests are failing resulting in a failing build - can you investigate?

It was the mock server setup for the tests
I added more checks before launching the server, and it were those checks not letting the server run without additional env variables that would ensure the checks I added passed
